src/stk/itk_bridge/itk_io.cpp

```diff
@@ -0,0 +1,304 @@
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::INT: {
+        using PixelType = itk::Vector<int, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::ULONG: {
+        using PixelType = itk::Vector<unsigned long, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::LONG: {
+        using PixelType = itk::Vector<long, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::ULONGLONG: {
+        using PixelType = itk::Vector<unsigned long long, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::LONGLONG: {
+        using PixelType = itk::Vector<long long, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::FLOAT: {
+        using PixelType = itk::Vector<float, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::DOUBLE: {
+        using PixelType = itk::Vector<double, Components>;
+        using ImageType = itk::Image<PixelType, Dimension>;
+        return read<ImageType>(file_name);
+    }
+    case itk::ImageIOBase::UNKNOWNCOMPONENTTYPE:
+    default:
+        FATAL() << "Cannot read image '" << &file_name << "': "
+                << "unknown component type";
+    }
+}
+
+
+/*!
+ * \brief Read a vectorial image of unspecified type with ITK.
+ */
+template<unsigned int Dimension>
+stk::Volume read_vector_image(
+        const std::string& file_name,
+        const itk::ImageIOBase::IOComponentType component_type,
+        const unsigned int components
+        )
+{
+    switch (components)
+    {
+    case 1:
+        return read_vector_image<Dimension, 1>(file_name, component_type);
+    case 3:
+        return read_vector_image<Dimension, 3>(file_name, component_type);
+    default:
+        FATAL() << "Cannot read image '" << &file_name << "': "
+                << "unsupported number of components " << components;
+    }
+
+    FATAL() << "This line should be unreachable";
+}
+
+
+/*!
+ * \brief Read an unknown image with ITK.
+ */
+stk::Volume stk::read_itk_image(const std::string& file_name)
+{
+    itk::ImageIOBase::Pointer image_io = itk::ImageIOFactory::CreateImageIO(
+            file_name.c_str(),
+            itk::ImageIOFactory::ReadMode
+            );
+
+    if (!image_io) {
+        FATAL() << "Cannot read image '" << &file_name <<"': "
+                << "the format is unrecognised by ITK";
+    }
+
+    image_io->SetFileName(file_name);
+    image_io->ReadImageInformation();
```

Comment:
So if I understand it correctly, you open the image file several times? First for image info and then in `read` to read image data. Is there no better way or is it simply how ITK works?